### PR TITLE
Update contribution guide

### PR DESCRIPTION
**Pull request by stefanv**
Specifically address review concerns, and what should happen with larger
PRs.

### CODE REVIEW

File: doc/contributing.md [hunk 1]
```diff
@@ -28,3 +32,97 @@ If possible, provide:
...
+
+We don't like arguing about code style, and likely you don't
+either. Therefore, we use code formatters: black for Python, and
+Prettier for JavaScript.  Code is an art, and opinions differ of what
+looks good: we choose to spend our time writing correct, elegant code.
+
+### Reviews
+
+All code that goes into SkyPortal is reviewed by two team members (or,
+if the contributor is a team member, by at least one other team
```
Comment (dannygoldstein) [inline doc/contributing.md:52]: Who counts as a "team member"?

**Comment by dannygoldstein** [comment]
Can we add a section on CI? I.e., PRs don't get merged until you there's a green check mark?

File: doc/contributing.md (hunk 1 above)
Comment (stefanv) [reply doc/contributing.md:52]: Clarified.

File: doc/contributing.md [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
-2. Make the changes you wish to submit on a new branch
-3. Submit a pull request on GitHub.
+To make a code contribution to the project: Clone the SkyPortal
+repository, make the changes you wish to submit on a new branch,
+and submit a pull request (PR) on GitHub.
 
-The other developers will provide feedback, and you may push updates into the same branch (which will also update your pull request), until general consensus is achieved.
+The other developers will provide feedback, and you may push updates
+into the same branch (which will also update your pull request), until
+general consensus is achieved that it should be merged.
```
[the remark this answers is not in the document]
Comment (stefanv) [reply doc/contributing.md:17]: Reworded; please check.

**Comment by stefanv** [comment]
@dannygoldstein I mentioned the CI in the testing section, and could expand that once done with the GitHub Actions switch.  Let me know.

File: doc/contributing.md [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
 SkyPortal is released under the Modified BSD license, which means that you are allowed to modify the code for your own purposes, as long as you retain our [copyright notice](https://github.com/skyportal/skyportal/blob/master/LICENSE.txt).
 
 However, we would love to grow the SkyPortal community, and integrate improvements directly into our [code repository on GitHub](https://github.com/skyportal/skyportal).
 
-## Code Changes
-
-To make a code contribution to the project:
+## Including your changes
 
-1. Clone the SkyPortal repository
```
Comment (joelachance) [inline doc/contributing.md:11]: I've typically forked repositories when I contribute, would this be a good thing to add to step 1?

File: doc/contributing.md [hunk 4]
```diff
@@ -28,3 +33,106 @@ If possible, provide:
...
+
+- Comments should provide provide actionable feedback, and can come in
+  two flavors: suggestions and requirements.  Requirements are direct
+  instructions such as "Define the variable first", whereas
+  suggestions are softer: "Consider whether you'd like to reformat
+  this dictionary" or "Recommendation: prefer the shorthand form".
+
+- When comments are left on your PR, let the reviewer resolve their
+  comments once they are satisfied they've been adequately addressed
+  (unless you've done exactly as they asked).
```
Comment (joelachance) [inline doc/contributing.md:83]: I think these points are very thoughtful and a good inclusion. I wish more OSS projects would include a section like this.

File: doc/contributing.md (hunk 3 above)
Comment (stefanv) [reply doc/contributing.md:11]: Yes, thanks! Good catch.

File: doc/contributing.md [hunk 5]
```diff
@@ -28,3 +38,106 @@ If possible, provide:
...
+invaluable for improving code quality, and no author ever merges their
+own work.
+
+We ask all reviewers to be mindful that there is a human at the other
+side of the PR.  As such, consider the review process as a
+conversation aimed at getting the work mergeable as quickly as
+possible:
+
+- Do not nitpick in comments; add small fixes as suggestions (click
+  the `±` button) or push uncontroversial changes as a commit into
```
Comment (acrellin) [inline doc/contributing.md:77]: This just adds a suggested change, but doesn't actually commit it yet

File: doc/contributing.md [hunk 6]
```diff
@@ -28,3 +38,106 @@ If possible, provide:
...
+
+### Focused PRs
+
+In SkyPortal, we squash all PRs before merging them.  Because of this,
+it is okay to merge the master branch into your branch, instead of
+rebasing.
+
+The resulting squashed commit should deal with one topic only.  For
+example, if you find that, while implementing a new component, you
+also need to fix another, split that out into a separate PR.
```
Comment (acrellin) [inline doc/contributing.md:98]: Great

File: doc/contributing.md (hunk 5 above)
Comment (stefanv) [reply doc/contributing.md]: @acrellin I wasn't sure what you wanted to see here, but I added some language---take a look and let me know.

**Comment by acrellin** [comment]
@stefanv that's perfect, thanks. merging

**Pull request merged**